src/redis_cmd.cc

```diff
@@ -97,6 +97,68 @@ AuthResult AuthenticateUser(Connection *conn, Config* config, const std::string&
   return AuthResult::OK;
 }
 
+Status ParseTTL(const std::vector<std::string> &args, const std::unordered_map<std::string, bool>& white_list, int *result) {
+    int ttl = 0;
+    int64_t expire = 0;
+    bool last_arg = false;
+    for (size_t i = 0; i < args.size(); i++) {
+        last_arg = (i == args.size() - 1);
+        std::string opt = Util::ToLower(args[i]);
+        if (opt == "ex" && !ttl && !last_arg) {
+            auto parse_result = ParseInt<int>(args[++i], 10);
+            if (!parse_result) {
+                return Status(Status::RedisParseErr, errValueNotInteger);
+            }
+            ttl = *parse_result;
+            if (ttl <= 0) return Status(Status::RedisParseErr, errInvalidExpireTime);
+        } else if (opt == "exat"  && !ttl && !expire && !last_arg) {
+            auto parse_result = ParseInt<int64_t>(args[++i], 10);
+            if (!parse_result) {
+                return Status(Status::RedisParseErr, errValueNotInteger);
+            }
+            expire = *parse_result;
+            if (expire <= 0) return Status(Status::RedisParseErr, errInvalidExpireTime);
+        } else if (opt == "pxat"  && !ttl && !expire && !last_arg) {
+            auto parse_result = ParseInt<uint64_t>(args[++i], 10);
+            if (!parse_result) {
+                return Status(Status::RedisParseErr, errValueNotInteger);
+            }
+            uint64_t expire_ms = *parse_result;
+            if (expire_ms <= 0) return Status(Status::RedisParseErr, errInvalidExpireTime);
+            if (expire_ms < 1000) {
+                expire = 1;
+            } else {
+                expire = static_cast<int64_t>(expire_ms / 1000);
+            }
+        } else if (opt == "px"  && !ttl && !last_arg) {
+            int64_t ttl_ms = 0;
+            auto parse_result = ParseInt<int64_t>(args[++i], 10);
+            if (!parse_result) {
+                return Status(Status::RedisParseErr, errValueNotInteger);
+            }
+            ttl_ms = *parse_result;
+            if (ttl_ms <= 0) return Status(Status::RedisParseErr, errInvalidExpireTime);
+            if (ttl_ms > 0 && ttl_ms < 1000) {
+                ttl = 1;  // round up the pttl to second
+            } else {
+                ttl = static_cast<int>(ttl_ms / 1000);
+            }
+        } else if (white_list.find(opt) != white_list.end()) {
+            // pass
```

Comment:
If you want to use unordered_map, you can set `iter->second` to `true` here.

Reply:
Good idea, thanks